src/utils/WindowCloseWarnPopup.jsx

```diff
@@ -0,0 +1,26 @@
+import { Component } from 'react';
+
+class WindowCloseWarnPopup extends Component {
+  componentDidMount = () => {
+    window.addEventListener('beforeunload', this.keepOnPage);
+  };
+
+  componentWillUnmount = () => {
+    window.removeEventListener('beforeunload', this.keepOnPage);
+  };
+
+  keepOnPage = event => {
+    const message = 'Warning! Unsaved changes';
+
+    // eslint-disable-next-line no-param-reassign
+    event.returnValue = message;
```

Comment:
Custom messages don't seem to be supported that well. In that case let's just return `true` and not worry about the custom message.